modules/router/src/listener.ts

```diff
@@ -822,6 +822,226 @@ export async function setupListeners(
+        nonce: 1,
+        channelAddress: channelAddress.getValue(),
+        timeout: DEFAULT_CHANNEL_TIMEOUT.toString(),
+        alice,
+        bob,
+        balances: [],
+        processedDepositsA: [],
+        processedDepositsB: [],
+        assetIds: [],
+        defundNonces: [],
+        merkleRoot: HashZero,
+        latestUpdate: {} as any,
+        networkContext: {
+          chainId,
+          channelFactoryAddress: chainAddresses[chainId].channelFactoryAddress,
+          transferRegistryAddress: chainAddresses[chainId].transferRegistryAddress,
+        },
+        aliceIdentifier: routerSigner.publicIdentifier,
+        bobIdentifier: counterparty,
+        inDispute: false,
+      });
+    };
+
+    let senderChannel = senderChannelRes.getValue() as FullChannelState | undefined;
+    if (!senderChannel) {
+      const placeholder = await getEmptyChannel(from, chainId);
+      if (placeholder.isError) {
+        await messagingService.respondToAuctionMessage(inbox, Result.fail(placeholder.getError()!));
+        return;
+      }
+      senderChannel = placeholder.getValue();
+    }
+    let recipientChannel = recipientChannelRes.getValue() as FullChannelState | undefined;
+    if (!recipientChannel) {
+      const placeholder = await getEmptyChannel(recipient, chainId);
+      if (placeholder.isError) {
+        await messagingService.respondToAuctionMessage(inbox, Result.fail(placeholder.getError()!));
+        return;
+      }
+      recipientChannel = placeholder.getValue();
+    }
+    const feeRes = await calculateFeeAmount(
+      BigNumber.from(amount),
+      false, // receive exact amount, to be reviewed
+      assetId,
+      senderChannel,
+      recipientAssetId,
+      recipientChannel,
+      chainReader,
+      routerSigner.publicIdentifier,
+      logger,
+    );
+    if (feeRes.isError) {
+      logger.error({ error: feeRes.getError() }, "Error in calculateFeeAmount");
+      await messagingService.respondToAuctionMessage(
+        inbox,
+        Result.fail(
+          new QuoteError(QuoteError.reasons.CouldNotGetFee, {
+            feeError: jsonifyError(feeRes.getError()!),
+            recipient,
+            recipientChainId,
+            recipientAssetId,
+            assetId,
+            chainId,
+            sender: from,
+          }),
+        ),
+      );
+      return;
+    }
+    const { fee, amount: quoteAmount } = feeRes.getValue();
+    const quote = {
+      assetId,
+      amount: quoteAmount.toString(),
+      chainId,
+      routerIdentifier: routerSigner.publicIdentifier,
+      recipient,
+      recipientChainId,
+      recipientAssetId,
+      fee: fee.toString(),
+      expiry: (Date.now() + (getConfig().feeQuoteExpiry ?? DEFAULT_FEE_EXPIRY)).toString(), // valid for next 2 blocks
+    };
+    const toSign = hashTransferQuote(quote);
+    try {
+      const signature = await routerSigner.signMessage(toSign);
+      await messagingService.respondToAuctionMessage(
```

Comment:
Do we need to send the full quote so that we have the signature?

Reply:
Signing only the quote